File: async-scala-client/src/main/scala/io/swagger/client/model/Entity.scala
package io.swagger.client.model

import org.joda.time.DateTime
import java.util.UUID


case class Entity (
  name: Option[String],  // Entity name, official name (Razão Social)
role: Option[String],  // - &#39;transporter&#39; - &#39;sender&#39; # REMETENTE - &#39;dispatcher&#39; # EXPEDITOR - &#39;receiver&#39; # RECEIVER - &#39;addressee&#39; # DESTINATÁRIO - &#39;other&#39; # OUTRO 
_type: EntityType,
federalTaxId: String,  // CNPJ/CPF of Sales Buyer, if CPF pattern is &#39;[0-9]{11}&#39; if CNPJ pattern is &#39;[0-9]{14}&#39;
cityTaxId: Option[String],  // City Tax ID
stateTaxId: Option[String],  // State Tax ID
suframa: Option[String],  // Suframa ID
phone: Option[String],  // Entity Phone
taxRegime: Option[String],
email: Option[String],  // Entity Email
subjectToSRF1234: Option[Boolean]  // Companies subject to follow same rules of Government entities (Instrução Normativa: SRF1234)
)

File: javascript-closure-angular-client/API/Client/Entity.js
goog.provide('API.Client.Entity');

/**
 * @record
 */
API.Client.Entity = function() {}

/**
 * Entity name, official name (Razão Social)
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.name;

/**
 * - 'transporter' - 'sender' # REMETENTE - 'dispatcher' # EXPEDITOR - 'receiver' # RECEIVER - 'addressee' # DESTINATÁRIO - 'other' # OUTRO 
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.role;

/**
 * @type {!API.Client.EntityType}
 * @export
 */
API.Client.Entity.prototype.type;

/**
 * CNPJ/CPF of Sales Buyer, if CPF pattern is '[0-9]{11}' if CNPJ pattern is '[0-9]{14}'
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.federalTaxId;

/**
 * City Tax ID
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.cityTaxId;

/**
 * State Tax ID
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.stateTaxId;

/**
 * Suframa ID
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.suframa;

/**
 * Entity Phone
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.phone;

/**
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.taxRegime;

/**
 * Entity Email
 * @type {!string}
 * @export
 */
API.Client.Entity.prototype.email;

/**
 * Companies subject to follow same rules of Government entities (Instrução Normativa: SRF1234)
 * @type {!boolean}
 * @export
 */
API.Client.Entity.prototype.subjectToSRF1234;

/** @enum {string} */
API.Client.Entity.RoleEnum = { 
  transporter: 'transporter',
  sender: 'sender',
  dispatcher: 'dispatcher',
  receiver: 'receiver',
  addressee: 'addressee',
  other: 'other',
}
/** @enum {string} */
API.Client.Entity.TaxRegimeEnum = { 
  realProfit: 'realProfit',
  estimatedProfit: 'estimatedProfit',
  simplified: 'simplified',
  simplifiedOverGrossthreshold: 'simplifiedOverGrossthreshold',
  simplifiedEntrepreneur: 'simplifiedEntrepreneur',
  notApplicable: 'notApplicable',
  individual: 'individual',
}
